File: 6. Python/EX50.py
# Solicita a quantidade de pessoas
qtd = int(input("# Quantas pessoas voce vai digitar: "))

# Inicializa variáveis para controle
mai = 0
auxNom = ""
vetNom = ["" for i in range(qtd)]  # Vetor para nomes
vetIda = [0 for i in range(qtd)]   # Vetor para idades

# Coleta dados das pessoas
for i in range(qtd):
    print(f"\n# Dados da {i+1}a pessoa: ")

    vetNom[i] = input("# Nome: ")           # Coleta o nome
    vetIda[i] = int(input("# Idade: "))      # Coleta a idade

    # Verifica se a idade atual é maior que a máxima
    if i == 0 or mai < vetIda[i]:
        mai = vetIda[i]         # Atualiza a idade máxima
        auxNom = vetNom[i]      # Armazena o nome da pessoa mais velha

# Exibe o nome da pessoa mais velha
print(f"\n# PESSOA MAIS VELHA: {auxNom}")

File: 4. Java/EX50.java
package Apostila;

import java.util.Scanner;

public class EX50
{
    public static void main(String[] args) 
    {
        String auxNom = null;
        int qtd, mai = 0;

        // Lê a quantidade de pessoas
        try (Scanner ent = new Scanner(System.in)) 
        {
            System.out.print("# Quantas pessoas voce vai digitar: ");
            qtd = ent.nextInt();
            
            String[] vetNom = new String[qtd];
            int[] vetIda = new int[qtd];
            
            // Lê os dados das pessoas
            for (int i = 0; i < qtd; i++)
            {
                System.out.format("\n# Dados da %da pessoa:\n", i + 1);
                
                System.out.print("# Nome: ");
                vetNom[i] = ent.next();
                ent.nextLine();
                
                System.out.print("# Idade: ");
                vetIda[i] = ent.nextInt();
                
                // Verifica a maior idade
                if ((i == 0) || (mai < vetIda[i]))
                {
                    mai = vetIda[i];
                    auxNom = vetNom[i];
                }
            }
        }

        // Exibe a pessoa mais velha
        System.out.println("\n# PESSOA MAIS VELHA: " + auxNom);
    }
}
